pkg/reconciler/apis/permissionclaimlabel/permissionclaimlabel_reconcile.go

```diff
@@ -159,80 +163,94 @@ func (c *controller) reconcilePermissionClaims(ctx context.Context, apiBinding *
 		}
 	}
 
-	klog.V(2).Infof("Adding permission claims %v for APIBinding %s", addedPermissionClaims, fmt.Sprintf("%s|%s", lc, apiBinding.Name))
+	if len(addedPermissionClaims) != 0 {
+		klog.V(2).Infof("Adding permission claims %v for APIBinding %s|%s", addedPermissionClaims, lc, apiBinding.Name)
+	}
 
 	var allPatchErrors []error
-	for gvr, claims := range addedPermissionClaims {
+	// For all the things that are in the accepted claims we need to re-sync
+	for gvr, claims := range syncClaims {
```

Comment:
🔴 no. As commented above (and before) we cannot just take accepted claims and label blindly. We need a intersection with the actual claims of the export(s).

Reply:
I don't think it is that big of a deal except for correctness. There are no security implications AFACT. And as stated above, this has been the case the whole time, and I agree we need this, but we are designing the API to enable this now.

Reply:
We discussed this in Slack. The switch away from accepted claims will come in follow-up.

But I wonder whether we can and should restore the old loop over just `addedPermissionClaims` unioned with `removedPermissionClaim`? Why over `syncClaims`?

Reply:
agreed to follow up. Added a todo here need to update the epic to include this change